### ISSUE TITLE
Reusing Data Transformations

### ISSUE THREAD

**Issue by tbateson** [nicklauslittle/horseshoe-j#14]
Imagine there exists some Java code that I want to convert to horseshoe, this could be a method or a part of a method like such:
```java
String capitalize(final String s) {
  return s.isEmpty() ? s : Character.toUpperCase(s.charAt(0)) + s.substring(1);
}
```

One way to implement this logic in horseshoe is to use it in a map literal to reuse computed values:
```handlebars
{{#["capitalized": s.isEmpty() ? s : Character.toUpperCase(s.charAt(0)) + s.substring(1)]}}
This is an capitalized word: {{capitalized}}
{{/}}
```

However the logic above needs to be copied between different templates or even within templates. For this example the solution seems obvious, use a partial. But once more entries in the map are added using a partial makes less sense as you're not just computing one value but multiple. I'm looking for a feature that adds the ability to define a data transformation in the template with the main goal of reusability.

One possible solution is to expand the current partial declaration and expansion functionality.

Pros:
 - Reuse existing partials declaration and expansion syntax (mostly)
Cons:
 - Somewhat convoluted usage?

```handlebars
{{<capitalize(isEmpty() ? Character.toUpperCase(charAt(0)) + substring(1))}}
{{#["shoe", "ring", "sing"]}}
{{#>capitalize(.)}}
{{/}}
```

Implementation details:
 - Add `>` as a unary operator that expands partials
 - New syntax: `{{#>partial-name(argument)}}`, `{{<partial-name(expression)}}`
 - Declaring a partial with an expression requires that its block is empty (`{{<partial-name(expression)}}{{/}}`

**Comment by nicklauslittle** [comment]
This should be implemented in d2bf8545d6ee4d88a20036a0be4493d0764a3471.

The syntax is `{{name->expression}}` or `{{name()->expression}}` to create a named expression. (These tags qualify for stand-aloneness.) The expression can be used in later expressions using function-like syntax with an optional argument (`name()` to execute on the currently scoped object, `name(..)` to execute on the parent object). Named expressions have scope and are only allowed within the scope they are declared. They always take precedence over equivalent methods available at any scope. If a method is preferred over a named expression, it can be prefixed with `./` or `../`.

Root-level named expressions in each partial / template are inherited when the partial / template is called (`{{>f}}` includes all root-level expressions from the partial "f" at the current scope). For example,
```mustache
{{<a}}
  {{lower->toLowerCase()}}
{{/}}
{{lower->toString()}}
{{upper->toUpperCase()}}
{{#\"Original String\"}}
  {{>a}}
  {{upper() + \"-\" + lower()}}
{{/}}
```
should result in "  ORIGINAL STRING-original string", because the "lower" expression is overridden when the partial "a" is included on line 7.

**Comment by tbateson** [comment]
I like it! More functionality may be necessary but this satisfies the request.
